### PR TITLE
MGMT-7996 Adding proxy methods for infraenv

### PR DESCRIPTION
**Pull request by nshidlin**
- move idict and proxy classes `helper_classes/kube_helpers->helper_classes to used in REST cluster and infraenv classes
- Add update_proxy method to InfraEnv
- Add upddate_infra_env_proxy method to Cluster

### CODE REVIEW

File: discovery-infra/test_infra/helper_classes/proxy_params.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from test_infra.helper_classes.idict import IDict
+
+
+class Proxy(IDict):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/proxy_params.py:4]: Few things:

1. Keep class name and module name the same
2. If there is no complicated logic inside `Proxy` decorate it with dataclass
3. If it's a dataclass you don't need IDict and can simply `from dataclasses import dataclass, asdict`
4. Why do we need this class? `models.Proxy` gives us the same functionality

File: discovery-infra/test_infra/helper_classes/proxy_params.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        self.http_proxy = http_proxy
+        self.https_proxy = https_proxy
+        self.no_proxy = no_proxy
+
+    def as_dict(self) -> dict:
+        return {
+            "httpProxy": self.http_proxy,
+            "httpsProxy": self.https_proxy,
+            "noProxy": self.no_proxy,
+        }
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/proxy_params.py:29]:
```suggestion

@dataclass
class Proxy():
     http_proxy: str
     https_proxy: str
     no_proxy: str
        
     def asdict(self) -> dict:
        return asdict(self)
```

Comment (osherdp) [inline discovery-infra/test_infra/helper_classes/proxy_params.py:29]: nit: can you add a new line at the end of the file?
on Linux sometimes pipes have weird results when files don't have new lines at the bottom

File: discovery-infra/test_infra/helper_classes/infra_env.py [hunk 3]
```diff
@@ -124,4 +135,15 @@ def patch_discovery_ignition(self, ignition_info: str) -> str:
...
+    def update_proxy(self, proxy: Proxy) -> None:
+        self.update_config(proxy=deepcopy(proxy))
+        infra_env_update_params = models.InfraEnvUpdateParams(
+                proxy=models.Proxy(
+                    https_proxy=self._config.proxy.https_proxy if self._config.proxy.https_proxy else None,
+                    http_proxy=self._config.proxy.http_proxy if self._config.proxy.http_proxy else None,
+                    no_proxy=self._config.proxy.no_proxy if self._config.proxy.no_proxy else None
+                )
+        )
+        self.api_client.update_infra_env(infra_env_id=self.id, infra_env_update_params=infra_env_update_params)
```
Comment (osherdp) [inline discovery-infra/test_infra/helper_classes/infra_env.py:149]: can you add a blank line at the end of the file?

File: discovery-infra/test_infra/helper_classes/proxy_params.py (hunk 1 above)
Comment (nshidlin) [reply discovery-infra/test_infra/helper_classes/proxy_params.py:4]: 1. The problem is that `models.Proxy` is not `pickleable`  so doing a deepcopy can cause issues
2. Changing
3. Wanted to keep the same logic as the existing class

File: discovery-infra/test_infra/helper_classes/infra_env.py (hunk 3 above)
Comment (nshidlin) [reply discovery-infra/test_infra/helper_classes/infra_env.py]: done

File: discovery-infra/test_infra/helper_classes/proxy_params.py (hunk 1 above)
Comment (eliorerz) [reply discovery-infra/test_infra/helper_classes/proxy_params.py]: @nshidlin You can `proxy.to_dict()`. 
e.g.
```python
from assisted_service_client import models
proxy = models.Proxy(http_proxy="http_proxy", https_proxy="https_proxy", no_proxy="no_proxy")
print(proxy.to_dict())

---
{'http_proxy': 'http_proxy', 'https_proxy': 'https_proxy', 'no_proxy': 'no_proxy'}

```

File: discovery-infra/test_infra/helper_classes/infra_env.py [hunk 4]
```diff
@@ -120,4 +121,11 @@ def patch_discovery_ignition(self, ignition_info: str) -> str:
         self.api_client.patch_discovery_ignition(infra_env_id=self.id, ignition_info=ignition_info)
 
     def get_details(self) -> models.infra_env.InfraEnv:
-        return self.api_client.get_infra_env(infra_env_id=self.id)
\ No newline at end of file
+        return self.api_client.get_infra_env(infra_env_id=self.id)
+
+    def update_proxy(self, proxy: models.Proxy) -> None:
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/infra_env.py:126]: Can you also please change `cluster.set_proxy_values` to match the same signature?

File: discovery-infra/test_infra/helper_classes/infra_env.py [hunk 5]
```diff
@@ -120,4 +121,11 @@ def patch_discovery_ignition(self, ignition_info: str) -> str:
...
+        self.update_config(proxy=proxy)
+        infra_env_update_params = models.InfraEnvUpdateParams(
+                proxy=self._config.proxy
+        )
+        self.api_client.update_infra_env(infra_env_id=self.id, infra_env_update_params=infra_env_update_params)
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/infra_env.py:131]: PEP 8: W292 no newline at end of file

File: discovery-infra/test_infra/helper_classes/config/base_entity_config.py [hunk 6]
```diff
@@ -26,4 +26,3 @@ class BaseEntityConfig(_BaseConfig, ABC):
     base_dns_domain: str = None
     entity_name: BaseName = None
     proxy: models.Proxy = None
-
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/config/base_entity_config.py:29]: PEP 8: W292 no newline at end of file

**Pull request merged**